bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java

```diff
@@ -0,0 +1,230 @@
+            }
+        } else if (command instanceof NextPreviousType) {
+            if (command == NextPreviousType.NEXT) {
+                adbConnection.sendKeyEvent(KEY_EVENT_NEXT);
+            } else if (command == NextPreviousType.PREVIOUS) {
+                adbConnection.sendKeyEvent(KEY_EVENT_PREVIOUS);
+            }
+        } else if (command instanceof RewindFastforwardType) {
+            if (command == RewindFastforwardType.FASTFORWARD) {
+                adbConnection.sendKeyEvent(KEY_EVENT_MEDIA_FAST_FORWARD);
+            } else if (command == RewindFastforwardType.REWIND) {
+                adbConnection.sendKeyEvent(KEY_EVENT_MEDIA_REWIND);
+            }
+        } else {
+            logger.warn("Unknown media control command: {}", command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        var currentConfig = getConfigAs(AndroidDebugBridgeConfiguration.class);
+        config = currentConfig;
+        adbConnection.configure(currentConfig.ip, currentConfig.port);
+        updateStatus(ThingStatus.UNKNOWN);
+        connectionCheckerSchedule = scheduler.scheduleWithFixedDelay(this::checkConnection, 0,
+                currentConfig.refreshTime, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        var schedule = connectionCheckerSchedule;
+        if (schedule != null) {
+            schedule.cancel(true);
+            connectionCheckerSchedule = null;
+        }
+        adbConnection.disconnect();
+    }
+
+    public void checkConnection() {
+        var currentConfig = config;
+        if (currentConfig == null)
+            return;
+        logger.debug("Refresh device {} status", currentConfig.ip);
+        if (adbConnection.isConnected()) {
+            if (thing.getStatus() != ThingStatus.ONLINE)
+                updateStatus(ThingStatus.ONLINE);
+            refreshStatus();
+        } else {
+            if (thing.getStatus() != ThingStatus.OFFLINE) {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+            try {
+                adbConnection.connect();
+            } catch (AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceException e) {
+                logger.debug("Error connecting to device: {}", e.getMessage());
+                return;
+            }
+            if (adbConnection.isConnected()) {
+                updateStatus(ThingStatus.ONLINE);
+                refreshStatus();
+            }
+        }
+    }
+
+    private void refreshStatus() {
+        try {
+            handleCommandInternal(new ChannelUID(this.thing.getUID(), MEDIA_VOLUME_CHANNEL), RefreshType.REFRESH);
+        } catch (IOException | InterruptedException | AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceException
+                | AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceReadException e) {
+            logger.warn("Unable to refresh media volume: {}", e.getMessage());
```

Comment:
Should such errors also cause the thing status to go offline?

Reply:
It will allow to detect that the device is offline sooner, so yes I'll change it

Reply:
For my Fire TV Stick this warning is logged all the time. I think we should reduce logging level here. Otherwise it will be very annoying. Or I have to reduce logging to ERROR for this binding in my logging config.